examples/jsm/materials/MeshWboitMaterial.js

```diff
@@ -0,0 +1,340 @@
+			#include <envmap_vertex>
+			#include <fog_vertex>
+
+		}`,
+
+	fragmentShader: /* glsl */`
+
+		precision highp float;
+		precision highp int;
+
+		// MeshBasicMaterial
+
+		uniform vec3 diffuse;
+		uniform float opacity;
+
+		#ifndef FLAT_SHADED
+
+			varying vec3 vNormal;
+
+		#endif
+
+		#include <common>
+		#include <dithering_pars_fragment>
+		#include <color_pars_fragment>
+		#include <uv_pars_fragment>
+		#include <uv2_pars_fragment>
+		#include <map_pars_fragment>
+		#include <alphamap_pars_fragment>
+		#include <alphatest_pars_fragment>
+		#include <aomap_pars_fragment>
+		#include <lightmap_pars_fragment>
+		#include <envmap_common_pars_fragment>
+		#include <envmap_pars_fragment>
+		#include <fog_pars_fragment>
+		#include <specularmap_pars_fragment>
+		#include <logdepthbuf_pars_fragment>
+		#include <clipping_planes_pars_fragment>
+
+		// MeshWboitMaterial
+
+		uniform float renderStage;
+		uniform float weight;
+
+		void main() {
+
+			// MeshBasicMaterial
+
+			#include <clipping_planes_fragment>
+
+			vec4 diffuseColor = vec4( diffuse, opacity );
+
+			#include <logdepthbuf_fragment>
+			#include <map_fragment>
+			#include <color_fragment>
+			#include <alphamap_fragment>
+			#include <alphatest_fragment>
+			#include <specularmap_fragment>
+
+			ReflectedLight reflectedLight = ReflectedLight( vec3( 0.0 ), vec3( 0.0 ), vec3( 0.0 ), vec3( 0.0 ) );
+
+			// accumulation (baked indirect lighting only)
+
+			#ifdef USE_LIGHTMAP
+
+				vec4 lightMapTexel = texture2D( lightMap, vUv2 );
+				reflectedLight.indirectDiffuse += lightMapTexel.rgb * lightMapIntensity * RECIPROCAL_PI;
+
+			#else
+
+				reflectedLight.indirectDiffuse += vec3( 1.0 );
+
+			#endif
+
+			// modulation
+
+			#include <aomap_fragment>
+
+			reflectedLight.indirectDiffuse *= diffuseColor.rgb;
+
+			vec3 outgoingLight = reflectedLight.indirectDiffuse;
+
+			#include <envmap_fragment>
+			#include <output_fragment>
+			#include <tonemapping_fragment>
+			#include <encodings_fragment>
+			#include <fog_fragment>
+			#include <premultiplied_alpha_fragment>
+			#include <dithering_fragment>
+
+			// MeshWboitMaterial
+
+			if ( renderStage == ${ WboitStages.Acummulation.toFixed( 1 ) } ) {
+
+				vec4 accum = gl_FragColor.rgba;
+
+				#ifndef PREMULTIPLIED_ALPHA
+					accum.rgb *= accum.a;
+				#endif
+
+				float z = gl_FragCoord.z;
+
+				/* Equation #9 */
+				// float w = accum.a * clamp( 0.03 / ( 1e-5 + pow( abs( z ) / 200.0, 4.0 ) ), 0.01, 300.0 );
+				// gl_FragColor = vec4( accum.rgb, accum.a ) * w;
+
+				/* McGuire 10/2013 */
+				// float w = clamp( pow( ( accum.a * 8.0 + 0.01 ) * ( - z * 0.95 + 1.0 ), 3.0 ) * 1e3, 1e-2, 3e2 );
+				// gl_FragColor = vec4( accum.rgb, accum.a ) * w;
+
+				/* Stevinz, Adjustable Weight */
+				float scaleWeight = 0.7 + ( 0.3 * weight );
+				float w = clamp( pow( ( accum.a * 8.0 + 0.001 ) * ( - z * scaleWeight + 1.0 ), 3.0 ) * 1000.0, 0.001, 300.0 );
+				gl_FragColor = vec4( accum.rgb, accum.a ) * w;
```

Comment:
Isn't `vec4( accum.rgm, accum.a )` just `accum`?

Reply:
Indeed, good catch. Leftover from before I accounted for the `PREMULTIPLIED_ALPHA` flag. I have simplified that section of the shader code.